recipes/server.rb

```diff
@@ -117,11 +117,18 @@ def remove_job(job, servernode)
 end
 clientnodes = clientsearchnodes.values
 
-# Filter out clients not in our environment, if applicable.
+# Filter out clients in the wrong environments, if applicable.
 if servernode['rdiff-backup']['server']['restrict-to-own-environment']
+  filterenvs = servernode['chef_environment']
```

Comment:
This will set `filterenvs` to a string, but you later call `filterenvs.empty?` which will error out on a string.